### PR TITLE
[BUG FIX] TimedeltaIndex constructor raises an AttributeError.

### PR DESCRIPTION
**Pull request by skirui-source**
Fixes: https://github.com/rapidsai/cudf/issues/9829

This PR fixes `TimedeltaIndex` constructor invocation by handling `NaT` values replacement with `nulls`.

### CODE REVIEW

File: python/cudf/cudf/tests/test_timedelta.py
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_timedelta.py:1428]:
```suggestion
@pytest.mark.parametrize("dtype", utils.TIMEDELTA_TYPES)
@pytest.mark.parametrize("name", [None, "delta-index"])
def test_create_TimedeltaIndex(dtype, name):
    gdi = cudf.TimedeltaIndex(
        [1132223, 2023232, 342234324, 4234324], dtype=dtype, name=name
    )
    pdi = gdi.to_pandas()

    assert_eq(pdi, gdi)
```

`name` test can be parametrized into a single pytest

**Review by galipremsagar — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/core/column/timedelta.py
Comment (bdice) [inline python/cudf/cudf/core/column/timedelta.py:233]:
```suggestion
        na_value = np.timedelta64("NaT", self.time_unit)
```

File: python/cudf/cudf/core/column/timedelta.py
Comment (bdice) [inline python/cudf/cudf/core/column/timedelta.py:231]: Are `NaN` and `NaT` the same?
```suggestion
        """Return a copy with NaT values replaced with nulls."""
```

File: python/cudf/cudf/core/column/timedelta.py
Comment (bdice) [inline python/cudf/cudf/core/column/timedelta.py:230]: Is it appropriate for this to be copy-pasted from `DatetimeColumn`? I would consider making this a standalone function that is called in `column.py` where needed instead of having this method defined on both classes.

There are only three invocations of this function which all follow the pattern `col = col._make_copy_with_na_as_null()`. It seems like a free function would be better: `col = _make_copy_with_na_as_null(col)`.

**Review by bdice — CHANGES_REQUESTED** [review]

File: python/cudf/cudf/core/column/timedelta.py
Comment (skirui-source) [reply python/cudf/cudf/core/column/timedelta.py:231]: From google: For datetime64[ns] types, `NaT `represents missing values.

File: python/cudf/cudf/core/column/column.py
Comment (bdice) [inline python/cudf/cudf/core/column/column.py:1776]: Does this typing error still occur? If mypy passes with `# type: ignore` removed, then please remove this comment as well.

File: python/cudf/cudf/core/column/column.py
Comment (bdice) [inline python/cudf/cudf/core/column/column.py:315]: This needs to raise, otherwise it will error if called with an improper type. Raising early is better than attempting to use the undefined variable `na_value`.
```suggestion
        if np.issubdtype(self.dtype, np.timedelta64):
            na_value = np.timedelta64("NaT", self.time_unit)
        elif np.issubdtype(self.dtype, np.datetime64):
            na_value = np.datetime64("NaT", self.time_unit)
        else:
            raise ValueError("This type does not support replacing NaT with null.")
```

File: python/cudf/cudf/core/column/column.py
Comment (bdice) [inline python/cudf/cudf/core/column/column.py:309]: I would still prefer to see this as a free function, outside of a class (not part of `Column`) because it is not a total function of `Column` (it only acts on specific data types) and is only called in the context of another free function `as_column`. I would move this code right above `def as_column` here because that keeps its definition close to the places the function is called: https://github.com/rapidsai/cudf/blob/6ca1f20b632d53c6f31b93ac80cf9fce09228ce0/python/cudf/cudf/core/column/column.py#L1674

To limit the scope even further, we might consider defining this function inside the `as_column` function scope. What do you think?

**Review by bdice — CHANGES_REQUESTED** [review]
Comments attached.

File: python/cudf/cudf/core/column/column.py
Comment (skirui-source) [reply python/cudf/cudf/core/column/column.py]: @bdice Following our call, I have implem. `_make_copy_replacing_NaT_with_null()` as a free function in `column.py`, all tests are currently passing without the Type Annotations- being specified so I left it as such. 

Because mypy complained about type-mismatches, otherwise (We can sync on a call to demonstrate)

**Review by bdice — APPROVED** [review]

File: python/cudf/cudf/core/column/column.py
Comment (bdice) [reply python/cudf/cudf/core/column/column.py]: Thanks! Your solution here is good.

**Comment by bdice** [comment]
@galipremsagar This can be merged once you approve. Thanks!

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**
